tools/lambda-promtail/lambda-promtail/promtail.go

```diff
@@ -198,6 +198,12 @@ func (c *promtailClient) send(ctx context.Context, buf []byte) (int, error) {
 		req.Header.Set("Authorization", "Bearer "+bearerToken)
 	}
 
+	if len(extraHeaders) > 0 {
+		for key, value := range extraHeaders {
+			req.Header.Set(key, value)
+		}
+	}
+
```

Comment:
I'm am wondering if, to avoid any potential unexpected issues, we should include some validation here. For example, not allowing overwriting of the headers we set earlier in this function, and logging if we overwrite an existing header value (such as someone not noticing that in their configured extra headers they're setting `SOME-EXTRA-HEADER=a, ... , SOME-EXTRA-HEADER=b`.

Reply:
I implemented a simple check to see if a key has already been set and added a warning for the user to check its env vars: https://github.com/grafana/loki/pull/11883/commits/6a8a19d68626e976e0a0bfe56b1b9825d4dc928d

Not sure about the logging part since I could not really find a clear way the logging is implemented in the code. Maybe you could have a closer look at that.

Reply:
@trc-ikeskin thanks! this is exactly what I had in mind. Logging in this context just meant fmt.Println or using the logger which you've done.

I still think we should guard against people overwriting the headers that are set earlier in this same function, such as `Authorization` or `X-Scope-OrgID`. We can do that here or in a follow up PR. Up to you.